pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -1009,8 +1012,12 @@ CompletableFuture<Void> subscribeAsync(String topicName, int numberPartitions) {
                     new PulsarClientException.AlreadyClosedException("Topic name not valid"));
         }
         String fullTopicName = topicNameInstance.toString();
-        if (consumers.containsKey(fullTopicName)
-                || partitionedTopics.containsKey(topicNameInstance.getPartitionedTopicName())) {
+        if (consumers.containsKey(fullTopicName)) {
+            return FutureUtil.failedFuture(
+                    new PulsarClientException.AlreadyClosedException("Already subscribed to " + topicName));
```

Comment:
It seems surprising to use `AlreadyClosedException` for this purpose. I can see that this is already used in this class a few lines below.

Reply:
>  It seems surprising to use AlreadyClosedException for this purpose. I can see that this is already used in this class a few lines below.

I think is is ok. These errors will be thrown at the following behaviors:
- When starting a partitioned consumer or a pattern consumer, the consumer may fail to subscribe due to this error, and the consumer will be closed soon. So it is ok to get an `AlreadyClosedException` at this case.
- After adding/removing partitions/topics, the error will only be used in the client library internally, it is strange.

How about changing it to `ConsumerBusyException`?

Reply:
I guess it's fine to keep it as-is since there's existing exceptions using AlreadyClosedException for this purpose. Changing the exception type could be a breaking change.